### PR TITLE
Add podman driver

### PR DESCRIPTION
**Pull request by medyagh**
- add vm-driver=podman closes https://github.com/kubernetes/minikube/issues/6446
closes https://github.com/kubernetes/minikube/issues/4390
- podman needs sudo explicitly .
- tested with 
```
 sudo podman version
Version:            1.7.0
RemoteAPI Version:  1
Go Version:         go1.11.6
OS/Arch:            linux/amd64

Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
```

- organize constants files
- add min required and suggested docker and podman versions
- added scripts for podman integration tests
- podman volume to be created inside  ./minikube/machine/name/var

### CODE REVIEW

File: pkg/drivers/kic/oci/oci.go [hunk 1]
```diff
@@ -63,6 +61,11 @@ func CreateContainerNode(p CreateParams) error {
 		// label the node with the role ID
 		"--label", fmt.Sprintf("%s=%s", nodeRoleKey, p.Role),
 	}
+	// Podman does not allow setting resources when cgroup is specified
+	// exact podman error: "Error: invalid configuration, cannot set resources with rootless containers not using cgroups v2 unified mode"
+	if p.OCIBinary != Podman {
+		runArgs = append(runArgs, fmt.Sprintf("--cpus=%s", p.CPUs), fmt.Sprintf("--memory=%s", p.Memory))
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: If possible, we should emit a warning whenever the users preferences are ignored.

File: pkg/drivers/kic/oci/oci.go [hunk 2]
```diff
@@ -159,11 +166,23 @@ func HostPortBinding(ociBinary string, ociID string, contPort int) (int, error)
 	if err := PointToHostDockerDaemon(); err != nil {
 		return 0, errors.Wrap(err, "point host docker-daemon")
 	}
-	cmd := exec.Command(ociBinary, "inspect", "-f", fmt.Sprintf("'{{(index (index .NetworkSettings.Ports \"%d/tcp\") 0).HostPort}}'", contPort), ociID)
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return 0, errors.Wrapf(err, "getting host-bind port %d for container ID %q, output %s", contPort, ociID, out)
+	var out []byte
+	var err error
+	if ociBinary == Podman {
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: These if statements are substantial enough that we should probably have it use separate Podman & Docker helper functions.

File: pkg/minikube/command/kic_runner.go [hunk 3]
```diff
@@ -150,22 +151,34 @@ func (k *kicRunner) Copy(f assets.CopyableFile) error {
...
+		return errors.Wrapf(err, "converting permissions %s to integer", f.GetPermissions())
 	}
 
 	// Rely on cp -a to propagate permissions
 	if err := os.Chmod(assetName, os.FileMode(perms)); err != nil {
 		return errors.Wrapf(err, "chmod")
 	}
-	if out, err := exec.Command(k.ociBin, "cp", "-a", assetName, destination).CombinedOutput(); err != nil {
-		return errors.Wrapf(err, "error copying %s into node, output: %s", f.GetAssetName(), string(out))
+	if k.ociBin == oci.Podman { // Podman cp command doesn't match docker and doesn't have -a
```
Comment (tstromberg) [inline pkg/minikube/command/kic_runner.go]: split out a podman and docker helper function for this.

File: pkg/minikube/registry/drvs/podman/podman.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		OCIBinary:     oci.Podman,
+		APIServerPort: mc.Nodes[0].Port,
+	})
+
+}
+
+func status() registry.State {
+	_, err := exec.LookPath(oci.Podman)
+	if err != nil {
+		return registry.State{Error: err, Installed: false, Healthy: false, Fix: "Podman is required.", Doc: "https://minikube.sigs.k8s.io/docs/reference/drivers/kic/"}
```
Comment (tstromberg) [inline pkg/minikube/registry/drvs/podman/podman.go]: URL should be /drivers/podman/

Comment (medyagh) [reply pkg/minikube/registry/drvs/podman/podman.go]: done

File: pkg/drivers/kic/oci/oci.go (hunk 2 above)
Comment (medyagh) [reply pkg/drivers/kic/oci/oci.go]: done

File: pkg/minikube/command/kic_runner.go (hunk 3 above)
Comment (medyagh) [reply pkg/minikube/command/kic_runner.go]: done

File: pkg/drivers/kic/oci/oci.go (hunk 1 above)
Comment (medyagh) [reply pkg/drivers/kic/oci/oci.go]: done

File: hack/jenkins/common.sh [hunk 5]
```diff
@@ -320,7 +320,7 @@ echo ">> Running go test2json"
 go tool test2json -t < "${TEST_OUT}" > "${JSON_OUT}" || true
 echo ">> Installing gopogh"
 cd /tmp
-GO111MODULE="on" go get -u github.com/medyagh/gopogh@v0.0.17 || true
+GO111MODULE="off" go get -u github.com/medyagh/gopogh@v0.0.17 || true
```
Comment (tstromberg) [inline hack/jenkins/common.sh]: Remove unrelated change

File: pkg/drivers/kic/oci/oci.go [hunk 6]
```diff
@@ -175,11 +194,35 @@ func HostPortBinding(ociBinary string, ociID string, contPort int) (int, error)
 
 // ContainerIPs returns ipv4,ipv6, error of a container by their name
 func ContainerIPs(ociBinary string, name string) (string, string, error) {
+	if ociBinary == Podman {
+		return podomanContinerIP(name)
+	}
+	return dockerContainerIP(name)
+}
+
+func podomanContinerIP(name string) (string, string, error) {
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: This function name has strange spelling.

File: pkg/drivers/kic/oci/oci.go [hunk 7]
```diff
@@ -175,11 +194,35 @@ func HostPortBinding(ociBinary string, ociID string, contPort int) (int, error)
...
+	cmd := exec.Command(Podman, "inspect",
+		"-f", "{{.NetworkSettings.IPAddress}}",
+		name)
+	out, err := cmd.CombinedOutput()
+	output := string(out)
+	output = strings.Trim(output, "\n")
+	output = strings.Trim(output, " ")
+	if err != nil {
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: Check for err before manipulating the string

File: pkg/drivers/kic/oci/oci.go [hunk 8]
```diff
@@ -175,11 +194,35 @@ func HostPortBinding(ociBinary string, ociID string, contPort int) (int, error)
...
+}
+
+func podomanContinerIP(name string) (string, string, error) {
+	cmd := exec.Command(Podman, "inspect",
+		"-f", "{{.NetworkSettings.IPAddress}}",
+		name)
+	out, err := cmd.CombinedOutput()
+	output := string(out)
+	output = strings.Trim(output, "\n")
+	output = strings.Trim(output, " ")
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: Use strings.TrimSpace() to remove newline and spaces.

File: pkg/minikube/bootstrapper/bsutil/kubelet_test.go [hunk 9]
```diff
@@ -99,15 +102,15 @@ ExecStart=/var/lib/minikube/binaries/v1.17.2/kubelet --authorization-mode=Webhoo
 					},
 				},
 			},
-			expected: `[Unit]
+			expected: fmt.Sprintf(`[Unit]
```
Comment (tstromberg) [inline pkg/minikube/bootstrapper/bsutil/kubelet_test.go]: Since this is effectively a change detector test, please put in the raw expected content rather than manipulating it with fmt.Sprintf. 

Otherwise it's too difficult for readers to understand what really is expected.

File: pkg/minikube/bootstrapper/bsutil/testdata/v1.11/containerd-api-port.yaml [hunk 10]
```diff
@@ -6,11 +6,11 @@ api:
   bindPort: 12345
   controlPlaneEndpoint: localhost
 kubernetesVersion: v1.11.0
-certificatesDir: /var/lib/minikube/certs
+certificatesDir: /usr/local/minikube/certs
 networking:
   serviceSubnet: 10.96.0.0/12
 etcd:
-  dataDir: /var/lib/minikube/etcd
+  dataDir: /usr/local/minikube/etcd
```
Comment (tstromberg) [inline pkg/minikube/bootstrapper/bsutil/testdata/v1.11/containerd-api-port.yaml]: Please keep user data in /var/lib/minikube if possible.

If not: Moving the etcd directory means we lose the previous cluster state. If we're going to move it, we'll need to have code that relocates the old directory or provides a symbolic link.

File: pkg/minikube/registry/drvs/docker/docker.go [hunk 11]
```diff
@@ -57,17 +60,33 @@ func configure(mc config.MachineConfig) interface{} {
...
+		return registry.State{Error: err, Installed: false, Healthy: false, Fix: "Docker is required.", Doc: "https://minikube.sigs.k8s.io/docs/reference/drivers/docker/"}
 	}
-	// Allow no more than 2 seconds for querying state
-	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+
+	cmd := exec.Command(oci.Docker, "version", "-f", "'{{.Server.Version}}'")
+	o, err := cmd.CombinedOutput()
+	output := string(o)
+	if err != nil {
+		return registry.State{Error: err, Installed: true, Healthy: false, Fix: "Cant verify mininim required version for podman . See podman website for installation guide.", Doc: "https://docs.docker.com/"}
```
Comment (tstromberg) [inline pkg/minikube/registry/drvs/docker/docker.go]: Check spelling of minimum.

File: pkg/minikube/registry/drvs/podman/podman.go [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+		return registry.State{Error: err, Installed: true, Healthy: false, Fix: "Cant verify mininim required version for podman . See podman website for installation guide.", Doc: "https://podman.io/getting-started/installation.html"}
+	}
+
+	v, err := semver.Make(output)
+	if err != nil {
+		return registry.State{Error: err, Installed: true, Healthy: false, Fix: "Cant verify mininim required version for podman . See podman website for installation guide.", Doc: "https://podman.io/getting-started/installation.html"}
+	}
+
+	if v.LT(constants.MinReqPodmanVer) {
+		glog.Warningf("Warning ! mininim required version for podman is %s. your version is %q. minikube might not work. use at your own risk. To install latest version please see https://podman.io/getting-started/installation.html ", constants.MinReqPodmanVer.String(), v.String())
```
Comment (tstromberg) [inline pkg/minikube/registry/drvs/podman/podman.go]: Check spelling of minimum.

File: pkg/minikube/registry/drvs/podman/podman.go [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+}
+
+func status() registry.State {
+	_, err := exec.LookPath(oci.Podman)
+	if err != nil {
+		return registry.State{Error: err, Installed: false, Healthy: false, Fix: "Podman is required.", Doc: "https://minikube.sigs.k8s.io/docs/reference/drivers/podman/"}
+	}
+
+	cmd := exec.Command(oci.Podman, "version", "-f", "{{.Version}}")
```
Comment (tstromberg) [inline pkg/minikube/registry/drvs/podman/podman.go]: Add a context timeout.

File: pkg/minikube/vmpath/constants.go [hunk 14]
```diff
@@ -24,7 +24,7 @@ const (
 	// GuestEphemeralDir is the path where ephemeral data should be stored within the VM
 	GuestEphemeralDir = "/var/tmp/minikube"
 	// GuestPersistentDir is the path where persistent data should be stored within the VM (not tmpfs)
-	GuestPersistentDir = "/var/lib/minikube"
+	GuestPersistentDir = "/usr/local/minikube"
```
Comment (tstromberg) [inline pkg/minikube/vmpath/constants.go]: /usr is on /tmpfs in the current ISO, which means it can't be used for persistent storage without changing it.

Comment (medyagh) [reply pkg/minikube/vmpath/constants.go]: what would be a better place for it ?

**Comment by medyagh** [comment]
suprisingly this PR is makes minikube 2 seconds faster ! :D

File: pkg/minikube/bootstrapper/bsutil/kubelet_test.go (hunk 9 above)
Comment (medyagh) [reply pkg/minikube/bootstrapper/bsutil/kubelet_test.go]: done

File: cmd/minikube/cmd/start.go [hunk 15]
```diff
@@ -812,16 +812,18 @@ func validateFlags(cmd *cobra.Command, drvName string) {
...
 		}
 		if cmd.Flags().Changed(memory) {
-			out.WarningT("The 'none' driver does not respect the --memory flag")
+			out.WarningT("The '{{.name}}' driver does not respect the --memory flag", out.V{"name": drvName})
+		}
+	}
+
+	if driver.BareMetal(drvName) {
+		if viper.GetString(config.MachineProfile) != constants.DefaultMachineName {
+			exit.WithCodeT(exit.Config, "The 'none' driver does not support multiple profiles: https://minikube.sigs.k8s.io/docs/reference/drivers/none/")
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: use drvName here instead of hardcoding none.

File: pkg/drivers/kic/oci/oci.go [hunk 16]
```diff
@@ -64,6 +63,23 @@ func CreateContainerNode(p CreateParams) error {
 		"--label", fmt.Sprintf("%s=%s", nodeRoleKey, p.Role),
 	}
 
+	// volume path in minikube home folder to mount to /var
+	hostVarVolPath := filepath.Join(localpath.MiniPath(), "machines", p.Name, "var")
+	if err := os.MkdirAll(hostVarVolPath, 0777); err != nil {
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: Is 0777 necessary? The permissions seems extremely broad. Prefer 0700 or 0711 if possible.

File: pkg/drivers/kic/oci/oci.go [hunk 17]
```diff
@@ -64,6 +63,23 @@ func CreateContainerNode(p CreateParams) error {
...
+		return errors.Wrapf(err, "create var dir %s", hostVarVolPath)
+	}
+
+	if p.OCIBinary == Podman { // enable execing in /var
+		// podman mounts var/lib with no-exec by default  https://github.com/containers/libpod/issues/5103
+		runArgs = append(runArgs, "--volume", fmt.Sprintf("%s:/var:exec", hostVarVolPath))
+	}
+	if p.OCIBinary == Docker {
+		runArgs = append(runArgs, "--volume", fmt.Sprintf("%s:/var", hostVarVolPath))
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: Could we limit this so that it is only mounted this way on Linux? It potentially offers poor performance when a VM is involved.

File: pkg/drivers/kic/oci/oci.go [hunk 18]
```diff
@@ -175,11 +207,34 @@ func HostPortBinding(ociBinary string, ociID string, contPort int) (int, error)
 
 // ContainerIPs returns ipv4,ipv6, error of a container by their name
 func ContainerIPs(ociBinary string, name string) (string, string, error) {
+	if ociBinary == Podman {
+		return podomanConttainerIP(name)
+	}
+	return dockerContainerIP(name)
+}
+
+func podomanConttainerIP(name string) (string, string, error) {
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: - Add comment that mentions it returns an IPv4/IPv6 pair.
- Fix function name spelling.

File: pkg/drivers/kic/oci/oci.go [hunk 19]
```diff
@@ -175,11 +207,34 @@ func HostPortBinding(ociBinary string, ociID string, contPort int) (int, error)
...
+	cmd := exec.Command(Podman, "inspect",
+		"-f", "{{.NetworkSettings.IPAddress}}",
+		name)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "podman inspect ip %s", name)
+	}
+	output := string(out)
+	output = strings.TrimSpace(output)
```
Comment (tstromberg) [inline pkg/drivers/kic/oci/oci.go]: combine these two lines.

File: pkg/minikube/constants/constants.go [hunk 20]
```diff
@@ -20,99 +20,96 @@ import (
...
-// KubernetesReleaseBinaries are Kubernetes release binaries required for
-// kubeadm (kubelet, kubeadm) and the addon manager (kubectl)
-var KubernetesReleaseBinaries = []string{"kubelet", "kubeadm", "kubectl"}
+)
 
-// ImageCacheDir is the path to the image cache directory
-var ImageCacheDir = localpath.MakeMiniPath("cache", "images")
+var (
+	// MinReqPodmanVer is required the mininum version of podman to be installed for podman driver.
+	MinReqPodmanVer = semver.Version{Major: 1, Minor: 7, Patch: 0}
```
Comment (tstromberg) [inline pkg/minikube/constants/constants.go]: Move single-package constants into the package they are used in.

File: pkg/minikube/driver/driver.go [hunk 21]
```diff
@@ -106,6 +108,11 @@ func BareMetal(name string) bool {
 	return name == None || name == Mock
 }
 
+// NeedsRoot returns true if driver needs to run with root privileges
+func NeedsRoot(name string) bool {
+	return name == None || name == Podman
```
Comment (tstromberg) [inline pkg/minikube/driver/driver.go]: This is fine for a first iteration, but do know that is possible to use podman without root:

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/building_running_and_managing_containers/index#running-rootfull-or-rootless_building-running-and-managing-containers

I'm not sure what the fastest way to notice whether or not the system configuration requires root or not, however.

File: pkg/minikube/registry/drvs/docker/docker.go [hunk 22]
```diff
@@ -56,17 +60,36 @@ func configure(mc config.MachineConfig) (interface{}, error) {
...
-	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+
+	// Allow no more than 1 seconds for getting version
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, oci.Docker, "version", "-f", "'{{.Server.Version}}'")
+	o, err := cmd.CombinedOutput()
+	if err != nil {
+		return registry.State{Error: err, Installed: true, Healthy: false, Fix: "Cant verify minimum required version for docker . See docker website for installation guide.", Doc: "https://docs.docker.com/"}
```
Comment (tstromberg) [inline pkg/minikube/registry/drvs/docker/docker.go]: Add apostrophe to can't.

File: pkg/minikube/registry/drvs/podman/podman.go [hunk 23]
```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
```
Comment (tstromberg) [inline pkg/minikube/registry/drvs/podman/podman.go]: 2020

File: pkg/minikube/registry/drvs/podman/podman.go [hunk 24]
```diff
@@ -0,0 +1,96 @@
...
+
+	// Allow no more than 2 seconds for version command
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, oci.Podman, "version", "-f", "{{.Version}}")
+	o, err := cmd.CombinedOutput()
+	output := string(o)
+	if err != nil {
+		return registry.State{Error: err, Installed: true, Healthy: false, Fix: "Cant verify mininim required version for podman . See podman website for installation guide.", Doc: "https://podman.io/getting-started/installation.html"}
```
Comment (tstromberg) [inline pkg/minikube/registry/drvs/podman/podman.go]: apostrophes here too.

**Comment by tstromberg** [comment]
Just nits - this looks ready to merge otherwise.

File: .github/workflows/main.yml [hunk 25]
```diff
@@ -142,3 +142,37 @@ jobs:
       with:
         name: none_on_ubuntu_latest_report
         path: report
+  podman_ubuntu_18_04:
+      runs-on: ubuntu-18.04
```
[the remark this answers is not in the document]
Comment (medyagh) [reply .github/workflows/main.yml]: they dont have other than ubuntu

File: cmd/minikube/cmd/start.go [hunk 26]
```diff
@@ -812,16 +812,18 @@ func validateFlags(cmd *cobra.Command, drvName string) {
 	validateDiskSize()
 	validateMemorySize()
 
-	if driver.BareMetal(drvName) {
-		if viper.GetString(config.MachineProfile) != constants.DefaultMachineName {
-			exit.WithCodeT(exit.Config, "The 'none' driver does not support multiple profiles: https://minikube.sigs.k8s.io/docs/reference/drivers/none/")
-		}
-
+	if driver.NeedsRoot(drvName) { // both podman and none need root and they both cant specify resources
```
[the remark this answers is not in the document]
Comment (medyagh) [reply cmd/minikube/cmd/start.go]: sounds good

File: pkg/minikube/driver/driver.go (hunk 21 above)
Comment (braderhart) [reply pkg/minikube/driver/driver.go]: Can't wait for this!

**Comment by runlevel5** [comment]
@medyagh Would this vm-driver mode need docker engine?

When I try to start with podman vm driver, I got following error:


```
[root@orion tle]# /usr/local/bin/minikube --alsologtostderr start --vm-driver=podman
I0317 13:34:33.087019    6950 notify.go:125] Checking for updates...
I0317 13:34:33.466099    6950 start.go:252] hostinfo: {"hostname":"orion.dev","uptime":7049,"bootTime":1584405424,"procs":441,"os":"linux","platform":"fedora","platformFamily":"fedora","platformVersion":"32","kernelVersion":"5.6.0-0.rc5.git0.2.fc32.ppc64le","virtualizationSystem":"","virtualizationRole":"","hostid":"5632f077-29a6-48c4-9d05-933910ac9490"}
I0317 13:34:33.483490    6950 start.go:262] virtualization:
😄  minikube v1.8.2 on Fedora 32 (ppc64le)
I0317 13:34:33.483766    6950 driver.go:226] Setting default libvirt URI to qemu:///system
✨  Using the podman (experimental) driver based on existing profile
I0317 13:34:33.660334    6950 start.go:299] selected driver: podman
I0317 13:34:33.660370    6950 start.go:488] validating driver "podman" against &{Name:minikube KeepContext:false EmbedCerts:false MinikubeISO: Memory:2200 CPUs:2 DiskSize:20000 Driver:podman HyperkitVpnKitSock: HyperkitVSockPorts:[] DockerEnv:[] InsecureRegistry:[] RegistryMirror:[] HostOnlyCIDR:192.168.99.1/24 HypervVirtualSwitch: HypervUseExternalSwitch:false HypervExternalAdapter: KVMNetwork:default KVMQemuURI:qemu:///system KVMGPU:false KVMHidden:false DockerOpt:[] DisableDriverMounts:false NFSShare:[] NFSSharesRoot:/nfsshares UUID: NoVTXCheck:false DNSProxy:false HostDNSResolver:true HostOnlyNicType:virtio NatNicType:virtio KubernetesConfig:{KubernetesVersion:v1.17.3 ClusterName:minikube APIServerName:minikubeCA APIServerNames:[] APIServerIPs:[] DNSDomain:cluster.local ContainerRuntime:docker CRISocket: NetworkPlugin: FeatureGates: ServiceCIDR:10.96.0.0/12 ImageRepository: ExtraOptions:[] ShouldLoadCachedImages:true EnableDefaultCNI:false NodeIP: NodePort:0 NodeName:} Nodes:[{Name:m01 IP: Port:8443 KubernetesVersion:v1.17.3 ControlPlane:true Worker:true}] Addons:map[]}
<truncated>

# HERE - seems to me Docker is still needed
E0317 13:34:35.298230    6950 cache.go:106] Error downloading kic artifacts:  error loading image: 
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
I0317 13:34:35.298441    6950 start.go:245] acquiring machines lock for minikube: {Name:mka00e65579c2b557a802898fd1cf03ec4ab30a1 Clock:{} Delay:500ms Timeout:15m0s Cancel:<nil>}



I0317 13:34:35.298717    6950 start.go:249] acquired machines lock for "minikube" in 248.971µs
I0317 13:34:35.298737    6950 start.go:85] Skipping create...Using existing machine configuration
⌛  Reconfiguring existing host ...
I0317 13:34:35.298770    6950 fix.go:63] fixHost starting: m01
🔄  Starting existing podman container for "minikube" ...
I0317 13:34:35.500262    6950 fix.go:65] fixHost completed within 201.487044ms
I0317 13:34:35.500305    6950 start.go:73] releasing machines lock for "minikube", held for 201.570442ms
W0317 13:34:35.500514    6950 exit.go:101] Unable to start VM. Please investigate and run 'minikube delete' if possible: driver start: get kic state: get container minikube status: exit status 125

💣  Unable to start VM. Please investigate and run 'minikube delete' if possible: driver start: get kic state: get container minikube status: exit status 125

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

**Comment by braderhart** [comment]
@runlevel5 It looks like you're trying to used Podman on an existing cluster. Have you tried deploying a new cluster by specifying a different profile?

**Comment by runlevel5** [comment]
@braderhart 

Yes I have tried to remove the existing cluster yet gotten the same warning on docker:

```
[root@orion tle]# minikube delete
🙄  "minikube" profile does not exist, trying anyways.
💀  Removed all traces of the "minikube" cluster.
[root@orion tle]# minikube delete
🙄  "minikube" profile does not exist, trying anyways.
💀  Removed all traces of the "minikube" cluster.
[root@orion tle]# rm -rf /tmp/minikube*
[root@orion tle]# rm -rf /tmp/juju*
[root@orion tle]# /usr/local/bin/minikube --alsologtostderr start --driver=podman
I0317 22:40:42.091465   10546 notify.go:125] Checking for updates...
I0317 22:40:42.369365   10546 start.go:252] hostinfo: {"hostname":"orion.dev","uptime":39818,"bootTime":1584405424,"procs":443,"os":"linux","platform":"fedora","platformFamily":"fedora","platformVersion":"32","kernelVersion":"5.6.0-0.rc5.git0.2.fc32.ppc64le","virtualizationSystem":"","virtualizationRole":"","hostid":"5632f077-29a6-48c4-9d05-933910ac9490"}
I0317 22:40:42.381682   10546 start.go:262] virtualization:
😄  minikube v1.8.2 on Fedora 32 (ppc64le)
I0317 22:40:42.381757   10546 driver.go:226] Setting default libvirt URI to qemu:///system
✨  Using the podman (experimental) driver based on user configuration
I0317 22:40:42.570332   10546 start.go:299] selected driver: podman
I0317 22:40:42.570356   10546 start.go:488] validating driver "podman" against <nil>
I0317 22:40:42.570382   10546 start.go:494] status for podman: {Installed:true Healthy:true Error:<nil> Fix: Doc:}
W0317 22:40:42.700205   10546 start.go:852] Unable to query memory limits: get podman system info: exit status 125
I0317 22:40:42.700471   10546 start.go:863] Using suggested 2200MB memory alloc based on sys=-1MB, container=-1MB
I0317 22:40:42.700663   10546 cache.go:97] Beginning downloading kic artifacts
I0317 22:40:42.700707   10546 preload.go:84] Found local preload: /root/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v1-v1.17.3-docker-overlay2.tar.lz4
I0317 22:40:42.700720   10546 cache.go:41] Caching tarball of preloaded images
I0317 22:40:42.700747   10546 preload.go:113] Found /root/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v1-v1.17.3-docker-overlay2.tar.lz4 in cache, skipping download
I0317 22:40:42.700750   10546 cache.go:99] Downloading gcr.io/k8s-minikube/kicbase:v0.0.7@sha256:a6f288de0e5863cdeab711fa6bafa38ee7d8d285ca14216ecf84fcfb07c7d176 to local daemon
I0317 22:40:42.700792   10546 image.go:84] Writing gcr.io/k8s-minikube/kicbase:v0.0.7@sha256:a6f288de0e5863cdeab711fa6bafa38ee7d8d285ca14216ecf84fcfb07c7d176 to local daemon
I0317 22:40:42.700764   10546 cache.go:44] Finished downloading the preloaded tar for v1.17.3 on docker
I0317 22:40:42.701503   10546 profile.go:121] Saving config to /root/.minikube/profiles/minikube/config.json ...
I0317 22:40:42.701926   10546 lock.go:35] WriteFile acquiring /root/.minikube/profiles/minikube/config.json: {Name:mk270d1b5db5965f2dc9e9e25770a63417031943 Clock:{} Delay:500ms Timeout:1m0s Cancel:<nil>}

# ========
E0317 22:40:44.130208   10546 cache.go:106] Error downloading kic artifacts:  error loading image: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
# ========

I0317 22:40:44.130429   10546 start.go:245] acquiring machines lock for minikube: {Name:mka00e65579c2b557a802898fd1cf03ec4ab30a1 Clock:{} Delay:500ms Timeout:15m0s Cancel:<nil>}
I0317 22:40:44.130652   10546 start.go:249] acquired machines lock for "minikube" in 200.14µs
I0317 22:40:44.130674   10546 start.go:82] Provisioning new machine with config: {Name:minikube KeepContext:false EmbedCerts:false MinikubeISO: Memory:2200 CPUs:2 DiskSize:20000 Driver:podman HyperkitVpnKitSock: HyperkitVSockPorts:[] DockerEnv:[] InsecureRegistry:[] RegistryMirror:[] HostOnlyCIDR:192.168.99.1/24 HypervVirtualSwitch: HypervUseExternalSwitch:false HypervExternalAdapter: KVMNetwork:default KVMQemuURI:qemu:///system KVMGPU:false KVMHidden:false DockerOpt:[] DisableDriverMounts:false NFSShare:[] NFSSharesRoot:/nfsshares UUID: NoVTXCheck:false DNSProxy:false HostDNSResolver:true HostOnlyNicType:virtio NatNicType:virtio KubernetesConfig:{KubernetesVersion:v1.17.3 ClusterName:minikube APIServerName:minikubeCA APIServerNames:[] APIServerIPs:[] DNSDomain:cluster.local ContainerRuntime:docker CRISocket: NetworkPlugin: FeatureGates: ServiceCIDR:10.96.0.0/12 ImageRepository: ExtraOptions:[] ShouldLoadCachedImages:true EnableDefaultCNI:false NodeIP: NodePort:0 NodeName:} Nodes:[{Name:m01 IP: Port:8443 KubernetesVersion:v1.17.3 ControlPlane:true Worker:true}] Addons:map[]}
I0317 22:40:44.130758   10546 start.go:101] createHost starting for "minikube" (driver="podman")
I0317 22:40:44.290506   10546 start.go:137] libmachine.API.Create for "minikube" (driver="podman")
I0317 22:40:44.290596   10546 main.go:110] libmachine: Reading certificate data from /root/.minikube/certs/ca.pem
I0317 22:40:44.290649   10546 main.go:110] libmachine: Decoding PEM data...
I0317 22:40:44.290680   10546 main.go:110] libmachine: Parsing certificate...
I0317 22:40:44.290847   10546 main.go:110] libmachine: Reading certificate data from /root/.minikube/certs/cert.pem
I0317 22:40:44.290885   10546 main.go:110] libmachine: Decoding PEM data...
I0317 22:40:44.290909   10546 main.go:110] libmachine: Parsing certificate...
```

**Comment by afbjorklund** [comment]
See #7071 for the IBM POWER (ppc64le), rather than commenting on this closed PR
